src/components/AuthStoreSetter.js

```diff
@@ -8,7 +8,9 @@ import { authStore } from 'src/libs/state';
 function AuthStoreSetter() {
   const auth = useAuth();
 
-  useOnMount(() => authStore.update(_.set(['authContext'], auth)));
+  useOnMount(() => {
+    authStore.update(_.set(['authContext'], auth));
```

Comment:
I don't understand-- is this a cleanup function?

Reply:
This one is actually fine since `atom.update` returns void.
https://github.com/DataBiosphere/terra-ui/blob/471ba5ce639d3a11620482c238c8af928668da5b/src/libs/utils.ts#L37-L60

However, there's no need to rely on that behavior. This ensures that the effect function returns void.

Reply:
I'm sorry, I still don't get how this change ensures that it returns void.

Reply:
I guess just because `undefined` gets returned since there is not an explicit return, whereas in the shortcut notation `authStore.update(_.set(['authContext'], auth))` is returned?

Reply:
Previously, the effect function returned the result of `authStore.update`. Now, the effect function has no return.

Reply:
Thanks, I just figured it out. It's pretty obvious now how folks can make this mistake! Makes one want explicit returns.

Reply:
Fortunately, TypeScript identifies these issues and will complain:
```
Argument of type '() => Promise<void>' is not assignable to parameter of type 'EffectCallback'.
  Type 'Promise<void>' is not assignable to type 'void | Destructor'.
```